### ISSUE TITLE
Destructuring of t (for tape)

### ISSUE THREAD

**Issue by AugustinLF** [skovhus/jest-codemods#23]
I wrote all my tests by destructuring the t parameter of tape test function:
```
test('Test stuff', ({plan, equal, end}) => {
    plan(1);
    ok(true);
    end();
});
```
This style doesn't work with this codemod, however, if you're interested, I'd gladly work on a PR!

Edit: After looking a bit at the source, the easiest way would be to first transform "my style" to the conventional one you expect. And this could go to the ava/tape helper file, since it seems that ava works the same way.

**Comment by skovhus** [comment]
Never seen that pattern before, but of course completely valid way of doing it. : )

>  if you're interested, I'd gladly work on a PR!

You are more than welcome! 👍 

> Edit: After looking a bit at the source, the easiest way would be to first transform "my style" to the conventional one you expect. 

Yes, I think converting this early to a single value `t` and changing all arguments that are destructured to be prefixed with `t.` should be a good initial modification to the AST.

> And this could go to the ava/tape helper file, since it seems that ava works the same way.

Same thing for AVA yes.

Let me know if you need any help. 

If you haven't worked with codemods before I recommend www.astexplorer.net ... I've just copy pasted the tape transformer in there including all imports, see https://astexplorer.net/#/kPoyXfSmTf

**Comment by AugustinLF** [comment]
That's a lovely way to anwser issues! I worked a bit with jscodeshift, I hope I'll be able to make it work, I'll let you know!

**Comment by skovhus** [comment]
Thanks! 😄

**Comment by skovhus** [comment]
Published as 0.6.0. Thanks for contributing @AugustinLF 🤗
